### PR TITLE
fix: google-anyscale-vpc-firewall - Firewall Rule Names

### PR DESCRIPTION
**Pull request by brent-anyscale**
The firewall rule names were not properly getting pulled from the variable when pulling from the `predefined_firewall_rules`.

Additonal updates:
- E2E test added for Services where functional verification isn't possible
- Removed the unit test for the firewall as it was no longer maintained.

Changes to be committed:
	modified:   ../modules/google-anyscale-vpc-firewall/main.tf
	deleted:    ../modules/google-anyscale-vpc-firewall/test/terraform_google_anyscale_vpc_firewall_test.go
	new file:   anyscale-service/service.py

## Pull request checklist

Please check if your PR fulfills the following requirements:
- [x] pre-commit has been run
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] All tests passing
- [x] Docs have been reviewed and added / updated if needed (for bug fixes / features)

## Pull Request Type

- [x] Bugfix
- [ ] New feature
- [ ] Refactoring (no functional changes)
- [ ] Documentation change
- [ ] Other (please describe):

## Does this introduce a breaking change?
- [ ] Yes
- [x] No

### CODE REVIEW

File: modules/google-anyscale-vpc-firewall/main.tf [hunk 1]
```diff
@@ -91,7 +91,7 @@ resource "google_compute_network_firewall_policy_rule" "ingress_allow_from_cidr_
   count   = local.ingress_from_cidr_blocks_enabled ? length(var.ingress_from_cidr_map) : 0
   project = var.anyscale_project_id
 
-  description     = lookup(var.ingress_from_cidr_map[count.index], "description", "Ingress Rule")
+  description     = try(var.predefined_firewall_rules[lookup(var.ingress_from_cidr_map[count.index], "rule", "")].description, "Ingress Rule")
```
Comment (hongchaodeng) [inline modules/google-anyscale-vpc-firewall/main.tf:94]: This is too long which makes it hard to understand...
How about use some locals to help?

```
locals {
  ingress_value = lookup(var.ingress_from_cidr_map[count.index], "rule", null)
  predefined_firewall_rule = local.ingress_value != null? var.predefined_firewall_rules[local.ingress_value]): null
}

description = local.predefined_firewall_rule != null? local.predefined_firewall_rule.description: "Ingress Rule"
```

**Review by hongchaodeng — APPROVED** [review]
LGTM in overall.
One nit. Leave it up to you for making the change.

File: modules/google-anyscale-vpc-firewall/main.tf (hunk 1 above)
Comment (brent-anyscale) [reply modules/google-anyscale-vpc-firewall/main.tf:94]: The trick here is that it's in a count loop—there could be 0, there could be 10, and each rule could (does) have a different description. I don't know how locals would be simpler in that situation.

Comment (brent-anyscale) [reply modules/google-anyscale-vpc-firewall/main.tf:94]: Keeping as is for now - will add a note to re-evaluate if we have a better way to handle this entire resource.

**Pull request merged**